node/src/Pos/Client/CLI/Params.hs

```diff
@@ -112,14 +114,38 @@ getNodeParams cArgs@CommonNodeArgs{..} NodeArgs{..} systemStart = do
         , ..
         }
 
-getTransportParams :: CommonNodeArgs -> NetworkConfig kademlia -> TransportParams
-getTransportParams args networkConfig = TransportParams { tpTcpAddr = tcpAddr }
-  where
-    tcpAddr = case ncTopology networkConfig of
-        TopologyBehindNAT{} -> TCP.Unaddressable
-        _ -> let (bindHost, bindPort) = bindAddress args
-                 (externalHost, externalPort) = externalAddress args
-                 tcpHost = BS8.unpack bindHost
-                 tcpPort = show bindPort
-                 tcpMkExternal = const (BS8.unpack externalHost, show externalPort)
-             in  TCP.Addressable $ TCP.TCPAddrInfo tcpHost tcpPort tcpMkExternal
+data NetworkTransportMisconfiguration =
+
+      -- | A bind address was not given.
+      MissingBindAddress
+
+      -- | An external address was not given.
+    | MissingExternalAddress
+
+      -- | An address was given when one was not expected (behind NAT).
+    | UnnecessaryAddress
+
+instance Show NetworkTransportMisconfiguration where
+    show MissingBindAddress     = "No network bind address given"
+    show MissingExternalAddress = "No external network address given"
+    show UnnecessaryAddress     = "Network address given when none was expected"
```

Comment:
I'm not so happy about this. I wrote it in the commit message

> The error messages aren't great. Why?
> You'd hope that we could do this in the command-line options
> parser (src/Pos/Client/CLI/Params.hs) but we can't because at that point
> we don't know whether the node is going to be in behind-NAT mode. I
> suppose we could juggle things around so that we parse the topology file
> before returning the SimpleNodeArgs. Will look into doing that if
> anyone insists.

The option parser isn't a monad so we can't make the `--address` and `--listen` options sensitive to the result of the `--topology` option... nor can we do `IO` in an option parser so that's all out the window.

Could parse the options in two passes.

Or we could just throw in the option names here, in this show instance. Feels wrong though.